app/server/lib/OIDCConfig.ts

```diff
@@ -314,13 +316,19 @@ export class OIDCConfig {
     });
   }
 
-  private _sendErrorPage(req: express.Request, res: express.Response, userFriendlyMessage?: string) {
+  private _sendErrorPage(
+    req: express.Request,
+    res: express.Response,
+    userFriendlyMessage?: string,
+    targetUrl: string = "/"
+  ) {
     return this._sendAppPage(req, res, {
       path: 'error.html',
       status: 500,
       config: {
         errPage: 'signin-failed',
-        errMessage: userFriendlyMessage
+        errMessage: userFriendlyMessage,
+        errTargetUrl: targetUrl,
```

Comment:
Maybe add some comment to explain the choice here? (sorry, I am not funny, sometimes I like when somebody explains me a joke 😅)

```suggestion
        // We redirect the user to the targetUrl, to let them refresh the page and therefore try to retrieve the resource
        // they originally asked (which in most cases results in another login attempt)
        errTargetUrl: targetUrl,
```

Reply:
Actually, it is already explained there:
https://github.com/gristlabs/grist-core/pull/1384/files#diff-3dab0a9106432a7a93722c32489b5f763381aee84436221ec6237370f3d6c203R791-R793

so it would be superfluous here.